### DISCUSSION TITLE
creating Phase

### DISCUSSION THREAD

**Discussion by kizzle4shaw** [PMEAL/OpenPNM#2701] [Q&A]
How do I create an oil phase?

**Comment by likeduck** [comment]
in v3, like this:

`Hg = op.phase.Mercury(network=pn)

phys = op.models.collections.physics.standard

Hg.add_model_collection(phys)`

And they have tutorial [here](https://openpnm.org/examples/tutorials/07_phases_and_mixtures.html)

**Reply by kizzle4shaw** [reply]
Mercury is already built in. I'm asking about a custom phase like oil which is not present in the phases collection.

**Reply by likeduck** [reply]
Although I didn 't try ( and that 's what I wanted to know ), I guess you need to give the `oil` various parameters, which included in the `phase` and `physic`.

**Reply by jgostick** [reply]
There are so many oils that its not possible to define a single one.  The idea is that you create a blank phase and add properties like viscosity by hand, or add correlations like temperature dependent viscosity using a pore- scale model such as a polynomial. We have some generic models in the math folder or you can write you own. The are examples in the docs: https://openpnm.org/_examples/tutorials.html

**Comment by likeduck** [comment]
Hi Prof @jgostick ！Hi @kizzle4shaw ！

I tried to define the fluid（oil）and two phase flow simulation is carried out.

First, define the oil. I‘m not yet skilled, hope to get advice.（I don 't understand the line of add phase.）

```
oil = op.phase.Phase(network=pn, name='oil')
oil['throat.contact_angle'] = 180
oil['pore.contact_angle'] = 180
oil['throat.surface_tension'] = 20e-3
oil['pore.surface_tension'] = 20e-3
oil['throat.viscosity'] = 0.8e-3
oil['pore.viscosity'] = 0.8e-3
# oil.add_model_collection(op.models.collections.phase.liquid)
oil.add_model_collection(op.models.collections.physics.basic)
oil.regenerate_models()
print(oil)
```

Then simulating.

(1) oil (define by previous step) and water.

![Figure_1](https://user-images.githubusercontent.com/117707506/221082527-bd99e021-dbda-4bbd-ab4e-28ea2e312091.png)

(2) air and water（Completely follow the [tutorial](https://openpnm.org/examples/applications/relative_permeability.html)）

![Kr_2](https://user-images.githubusercontent.com/117707506/221082563-13179aec-ca51-453a-b4a9-c439d06e0652.png)

In(1), Their values become 0 when they just intersect. And I changed those assignments many times, and the results did not change.

Could you please help me check the above code where is wrong？

Much Thanks

**Reply by JMCarmona10** [reply]
I have a question regarding the curves in this example: When modifying the properties of oil and water, did you observe any changes in the curve behavior? I'm specifically having difficulty when switching from a drainage process to an imbibition process. Any tips or guidance would be greatly appreciated.
